### PR TITLE
add changelogs

### PR DESCRIPTION
**Pull request by sinisastanic**

### CODE REVIEW

File: changelog.mdx [hunk 1]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
 icon: "sparkles"
 ---
 
+## 2024-02-23
+
+- `improved stability`: (not sure how to explain here)
```
Comment (RunOrVeith) [inline changelog.mdx:9]:
```suggestion
- `improved stability`: serverless agent and test execution
```

File: changelog.mdx [hunk 2]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+- `discover test cases`: our AI Agent will analyse your webpage and suggest multiple prompts which describe your test flows
```
Comment (RunOrVeith) [inline changelog.mdx:10]:
```suggestion
- `discover test cases`: our AI Agent will analyse your webpage and suggest multiple prompts
```

File: changelog.mdx (hunk 1 above)
Comment (Germandrummer92) [inline changelog.mdx:9]:
```suggestion
- `improved stability`: made sure our app is running smoothly without hick-ups even with multiple test report running in parallel
```

File: changelog.mdx (hunk 2 above)
Comment (Germandrummer92) [inline changelog.mdx:10]:
```suggestion
- `discover test cases`: our AI Agent will analyse your webpage and suggest multiple prompts which describe your user flows
```

File: changelog.mdx [hunk 3]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/discover-screen.png" alt="new discovery feature" />
```
Comment (Germandrummer92) [inline changelog.mdx:13]:
```suggestion
  <img src="/images/changelog/discover-screen.png" alt="new discovery feature" />
```

File: changelog.mdx [hunk 4]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+</Frame>
+
+- `approval workflow`: once our AI Agent has finished the discovery, you have an option to approve or discard provided suggestions.
+  Discarding will delete the draft, while approval will give the prompt to our AI Agent to start generating steps from it
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/approval-screen.png" alt="new discovery feature" />
```
Comment (Germandrummer92) [inline changelog.mdx:20]:
```suggestion
  <img src="/images/changelog/approval-screen.png" alt="new discovery feature" />
```

File: changelog.mdx [hunk 5]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+</Frame>
+
+## 2024-02-15
+
+- `test report generation speed up` : we are now running published test cases within a report in parallel
+- `auto fix`: you can now select which test cases our AI Agent should try to auto fix once they fail
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/auto-fix.png" alt="auto fix feature" />
```
Comment (Germandrummer92) [inline changelog.mdx:29]:
```suggestion
  <img src="/images/changelog/auto-fix.png" alt="auto fix feature" />
```

File: changelog.mdx [hunk 6]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+  - we replaced save rail with the functionalities that were inside it. Now we have 3 separate buttons
+
+  1. save button: to save changes made
+  2. run button: to run the selected test case
+  3. publish button
+
+  run button is also visible in on published test cases
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/save-rail-overhaul.png" alt="save rail overhaul" />
```
Comment (Germandrummer92) [inline changelog.mdx:43]:
```suggestion
  <img src="/images/changelog/save-rail-overhaul.png" alt="save rail overhaul" />
```

File: changelog.mdx [hunk 7]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+- `test report generation speed up` : we are now running published test cases within a report in parallel
+- `auto fix`: you can now select which test cases our AI Agent should try to auto fix once they fail
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/auto-fix.png" alt="auto fix feature" />
+</Frame>
+
+- save rail overhaul:
+
+  - we replaced save rail with the functionalities that were inside it. Now we have 3 separate buttons
```
Comment (RunOrVeith) [inline changelog.mdx:34]:
```suggestion
  we replaced save rail with the functionalities that were inside it. Now we have 3 separate buttons
```

File: changelog.mdx [hunk 8]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+## 2024-02-23
+
+- `improved stability`: (not sure how to explain here)
+- `discover test cases`: our AI Agent will analyse your webpage and suggest multiple prompts which describe your test flows
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/discover-screen.png" alt="new discovery feature" />
+</Frame>
+
+- `approval workflow`: once our AI Agent has finished the discovery, you have an option to approve or discard provided suggestions.
```
Comment (Germandrummer92) [inline changelog.mdx:16]:
```suggestion
- `approval workflow`: once our AI Agent has finished the discovery, you have an option to approve or discard the provided suggestions.
```

File: changelog.mdx (hunk 7 above)
Comment (RunOrVeith) [reply changelog.mdx:34]: maybe look at that one, but we don't need single bullet point within single bulelt point

File: changelog.mdx [hunk 9]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+  Discarding will delete the draft, while approval will give the prompt to our AI Agent to start generating steps from it
```
Comment (Germandrummer92) [inline changelog.mdx:17]:
```suggestion
  Discarding will delete the draft, while approval will start our AI Agent with the corresponding prompt
```

File: changelog.mdx [hunk 10]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+
+- save rail overhaul:
+
+  - we replaced save rail with the functionalities that were inside it. Now we have 3 separate buttons
+
+  1. save button: to save changes made
+  2. run button: to run the selected test case
+  3. publish button
+
+  run button is also visible in on published test cases
```
Comment (RunOrVeith) [inline changelog.mdx:40]:
```suggestion
 The `run` button is also visible in published test cases
```

File: changelog.mdx [hunk 11]
```diff
@@ -4,6 +4,45 @@ description: "All the nice stuff you asked for"
...
+## 2024-02-15
+
+- `test report generation speed up` : we are now running published test cases within a report in parallel
+- `auto fix`: you can now select which test cases our AI Agent should try to auto fix once they fail
+
+<Frame caption="screenshot 02/2024">
+  <img src="images/changelog/auto-fix.png" alt="auto fix feature" />
+</Frame>
+
+- save rail overhaul:
```
Comment (Germandrummer92) [inline changelog.mdx:32]:
```suggestion
- save button for test case drafts overhaul:
```

Comment (Germandrummer92) [reply changelog.mdx:32]: I dont think people know what a rail is

File: changelog.mdx (hunk 7 above)
Comment (Germandrummer92) [inline changelog.mdx:34]:
```suggestion
  - we replaced the save rail with multiple buttons that were hidden in a separate screen. Now we have 3 separate buttons
```

File: changelog.mdx (hunk 10 above)
Comment (Germandrummer92) [inline changelog.mdx:40]:
```suggestion
  run button is also visible on published test cases
```

**Review by RunOrVeith — COMMENTED** [review]
some small stuff

**Review by Germandrummer92 — APPROVED** [review]

**Review by RunOrVeith — APPROVED** [review]

**Pull request merged**